crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -500,39 +481,54 @@ where
     ///         .with_executor(TokioTaskExecutor::default())
     ///         .with_events(TestCanonStateSubscriptions::default())
     ///         .with_evm_config(evm)
-    ///         .into_registry(Default::default());
+    ///         .into_registry(Default::default(), EthApiBuild::build);
     ///
     ///     let eth_api = registry.eth_api();
     /// }
     /// ```
-    pub fn into_registry(
+    pub fn into_registry<EthApi, EthApiB>(
         self,
         config: RpcModuleConfig,
-    ) -> RethModuleRegistry<Provider, Pool, Network, Tasks, Events, EvmConfig> {
+        eth: EthApiB,
+    ) -> RpcRegistryInner<Provider, Pool, Network, Tasks, Events, EthApi>
+    where
+        EthApiB: FnOnce(&EthApiBuilderCtx<Provider, Pool, EvmConfig, Network, Tasks, Events>) -> EthApi
+            + 'static,
+    {
         let Self { provider, pool, network, executor, events, evm_config } = self;
-        RethModuleRegistry::new(provider, pool, network, executor, events, config, evm_config)
+        RpcRegistryInner::new(provider, pool, network, executor, events, config, evm_config, eth)
     }
 
     /// Configures all [`RpcModule`]s specific to the given [`TransportRpcModuleConfig`] which can
     /// be used to start the transport server(s).
     ///
     /// See also [`RpcServer::start`]
-    pub fn build(self, module_config: TransportRpcModuleConfig) -> TransportRpcModules<()> {
+    pub fn build<EthApi, EthApiB>(
+        self,
+        module_config: TransportRpcModuleConfig,
+        eth: EthApiB,
+    ) -> TransportRpcModules<()>
+    where
```

Comment:
can we add `build_ethereum` that calls this with the ethereum builder?

Reply:
not yet I think, since rpc isn't passed down from that level yet. or wdym?